### PR TITLE
`General`: Add tooltip to due date of exercise course card

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

#5029 redesigned the exercise header and added a tooltip to the dates of the exercise, when hovering over them. But an other place, namely the course card was left out.

### Description

This PR adds a short tooltip to the course exercise card that displays the due date, when hovering over it.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exercise with due date

1. Log in to Artemis
2. Navigate to the course
3. Hover over the due date in the exercise card
4. Make sure that you are redirected, even when clicking directly on the due date

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Bildschirm­foto 2022-12-09 um 11 39 13

### CODE REVIEW

**Review by evgeny1984 — DISMISSED** [review]
Tested on TS2, works as described, no issues. The code changes are also looking good.

**Review by ge65cer — CHANGES_REQUESTED** [review]
Good addition! 👍  However, the client tests failed.

**Review by stefanwaldhauser — APPROVED** [review]
Nice Tooltip :)

**Review by timo07k — DISMISSED** [review]
Tooltip works as expected. Tested on TS1.

**Review by LiaLism — DISMISSED** [review]
Tooltip worked as expected. Tested on ts1.

**Review by evgeny1984 — DISMISSED** [review]
Re-approved after minor changes. Tested on the ts1, works as expected.

File: src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ export class CourseExerciseRowComponent implements OnInit, OnDestroy, OnChanges
     dueDate?: dayjs.Dayjs;
     gradedStudentParticipation?: StudentParticipation;
 
+    routerLink: any[];
```
Comment (krusche) [inline src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts:45]: can we use `string[]` here?
(we generally want to avoid `any`)

**Pull request merged**